File: src/main/kotlin/Java/Biblioteca/Biblioteca.java
package Java.Biblioteca;

import java.util.ArrayList;
import java.util.List;

public class Biblioteca {
    private List<Libro> libros;
    private List<Usuario> usuarios;

    public Biblioteca() {
        libros = new ArrayList<>();
        usuarios = new ArrayList<>();
    }

    public void agregarLibro(Libro libro) {
        libros.add(libro);
        System.out.println("Libro agregado: " + libro);
    }

    public void registrarUsuario(Usuario usuario) {
        usuarios.add(usuario);
        System.out.println("Usuario registrado: " + usuario);
    }

    public void prestarLibro(int libroId, int usuarioId) {
        Libro libro = libros.stream().filter(l -> l.getId() == libroId).findFirst().orElse(null);
        Usuario usuario = usuarios.stream().filter(u -> u.getId() == usuarioId).findFirst().orElse(null);

        if (libro == null) {
            System.out.println("El libro con ID " + libroId + " no existe.");
            return;
        }
        if (usuario == null) {
            System.out.println("El usuario con ID " + usuarioId + " no existe.");
            return;
        }
        if (libro.isPrestado()) {
            System.out.println("El libro '" + libro.getTitulo() + "' ya está prestado.");
            return;
        }

        libro.setPrestado(true);
        System.out.println("Libro '" + libro.getTitulo() + "' prestado a " + usuario.getNombre() + ".");
    }

    public void devolverLibro(int libroId) {
        Libro libro = libros.stream().filter(l -> l.getId() == libroId).findFirst().orElse(null);

        if (libro == null) {
            System.out.println("El libro con ID " + libroId + " no existe.");
            return;
        }
        if (!libro.isPrestado()) {
            System.out.println("El libro '" + libro.getTitulo() + "' no estaba prestado.");
            return;
        }

        libro.setPrestado(false);
        System.out.println("Libro '" + libro.getTitulo() + "' devuelto.");
    }

    public void mostrarPrestamos() {
        libros.stream().filter(Libro::isPrestado).forEach(libro -> {
            System.out.println("Libro '" + libro.getTitulo() + "' está prestado.");
        });
    }
}

File: src/main/kotlin/Kotlin/Biblioteca/Biblioteca.kt
package Kotlin.Biblioteca

class Biblioteca {
    private val libros = mutableListOf<Libro>()
    private val usuarios = mutableListOf<Usuario>()
    private val prestamos = mutableMapOf<Int, Int>() // Map<LibroID, UsuarioID>

    fun agregarLibro(libro: Libro) {
        libros.add(libro)
        println("Libro agregado: $libro")
    }

    fun registrarUsuario(usuario: Usuario) {
        usuarios.add(usuario)
        println("Usuario registrado: $usuario")
    }

    fun prestarLibro(libroId: Int, usuarioId: Int) {
        val libro = libros.find { it.id == libroId }
        val usuario = usuarios.find { it.id == usuarioId }

        if (libro == null) {
            println("El libro con ID $libroId no existe.")
            return
        }
        if (usuario == null) {
            println("El usuario con ID $usuarioId no existe.")
            return
        }
        if (libro.prestado) {
            println("El libro '${libro.titulo}' ya está prestado.")
            return
        }

        libro.prestado = true
        prestamos[libroId] = usuarioId
        println("Libro '${libro.titulo}' prestado a ${usuario.nombre}.")
    }

    fun devolverLibro(libroId: Int) {
        val libro = libros.find { it.id == libroId }

        if (libro == null) {
            println("El libro con ID $libroId no existe.")
            return
        }
        if (!libro.prestado) {
            println("El libro '${libro.titulo}' no estaba prestado.")
            return
        }

        libro.prestado = false
        prestamos.remove(libroId)
        println("Libro '${libro.titulo}' devuelto.")
    }

    fun mostrarPrestamos() {
        prestamos.forEach { (libroId, usuarioId) ->
            val libro = libros.find { it.id == libroId }
            val usuario = usuarios.find { it.id == usuarioId }
            if (libro != null && usuario != null) {
                println("Libro '${libro.titulo}' prestado a ${usuario.nombre}.")
            }
        }
    }
}
